### PR TITLE
groups: shard some group definitions to subdirectories

### PR DESCRIPTION
**Pull request by spiffxp**
Related:
- Part of: https://github.com/kubernetes/k8s.io/issues/460
- Followup to: https://github.com/kubernetes/k8s.io/pull/2414

The actual group moves are broken down mostly one-group-per-commit with the exception of `k8s-infra-concorm-*`, `k8s-infra-staging-cluster-api-*` and `k8s-infra-staging-capi-*`.  I'll edit the description to summarize.

This is being done to prepare for delegation of group approval authority. Something living in root groups.yaml should be a very rare exception, and root group approvers should only really get notified of group additions or removals, to make sure the name is in line with policy.

This is part 1 of N, I'm trying to avoid a difficult-to-rebase PR hanging out while people add more groups or change memberships.

```
$ git log --pretty=oneline | grep "groups: mv" | awk '{print "- `" $6 "/groups.yaml` now contains: `" $4 "`"}' | sort
```

- `committee-steering/groups.yaml` now contains: `election`
- `sig-architecture/groups.yaml` now contains: `k8s-infra-code-organization`
- `sig-architecture/groups.yaml` now contains: `k8s-infra-conform-*`
- `sig-architecture/groups.yaml` now contains: `k8s-infra-staging-apisnoop`
- `sig-autoscaling/groups.yaml` now contains: `k8s-infra-staging-autoscaling`
- `sig-cluster-lifecycle/groups.yaml` now contains: `k8s-infra-staging-addon-manager`
- `sig-cluster-lifecycle/groups.yaml` now contains: `k8s-infra-staging-capi-*`
- `sig-cluster-lifecycle/groups.yaml` now contains: `k8s-infra-staging-cluster-addons`
- `sig-cluster-lifecycle/groups.yaml` now contains: `k8s-infra-staging-cluster-api-*`
- `sig-contributor-experience/groups.yaml` now contains: `github`
- `sig-contributor-experience/groups.yaml` now contains: `k8s-infra-rbac-slack-infra`
- `sig-contributor-experience/groups.yaml` now contains: `moderators`
- `sig-docs/groups.yaml` now contains: `blog`
- `sig-docs/groups.yaml` now contains: `k8s-staging-sig-docs`
- `sig-network/groups.yaml` now contains: `k8s-infra-push-cni`
- `sig-node/groups.yaml` now contains: `k8s-infra-push-cri-tools`
- `sig-release/groups.yaml` now contains: `k8s-infra-rbac-publishing-bot`
- `sig-scalability/groups.yaml` now contains: `k8s-infra-rbac-perdash`
- `sig-scalability/groups.yaml` now contains: `k8s-infra-rbac-triageparty-scalability`
- `sig-scalability/groups.yaml` now contains: `k8s-infra-sig-scalability-oncall`
- `sig-storage/groups.yaml` now contains: `k8s-infra-push-csi`
- `sig-storage/groups.yaml` now contains: `k8s-staging-sig-storage`
- `wg-k8s-infra/groups.yaml` now contains: `k8s-infra-artifact-admins`
- `wg-k8s-infra/groups.yaml` now contains: `k8s-infra-bigquery-admins`
- `wg-k8s-infra/groups.yaml` now contains: `k8s-infra-dns-admins`
- `wg-k8s-infra/groups.yaml` now contains: `k8s-infra-gcs-access-logs`
- `wg-k8s-infra/groups.yaml` now contains: `k8s-infra-rbac-cert-manager`
- `wg-k8s-infra/groups.yaml` now contains: `k8s-infra-rbac-k8s-io-canary`
- `wg-k8s-infra/groups.yaml` now contains: `k8s-infra-rbac-k8s-io-prod`

### CODE REVIEW

File: groups/groups.yaml [hunk 1]
```diff
@@ -138,36 +119,6 @@ groups:
...
-      - spiffxp@gmail.com
-      - spiffxp@google.com
-    managers:
-      - fejta@google.com
-      - ihor.dvoretskyi@gmail.com
-      - nikitaraghunath@gmail.com
-      - killen.bob@gmail.com
-
   # Every RBAC group should be added here.
   - email-id: gke-security-groups@kubernetes.io
```
Comment (ameukam) [inline groups/groups.yaml:123]: @spiffxp why not move it to `groups/wg-k8s-infra` ?

File: groups/groups.yaml [hunk 2]
```diff
@@ -336,63 +165,6 @@ groups:
...
-      - jprzychodzen@google.com
-      - jupblb@google.com
-      - maciejborsz@google.com
-      - marcelzieba@google.com
-      - mmatejczyk@google.com
-      - shyam123.jvs95@gmail.com
-      - wojtekt@google.com
-      - zawodny@google.com
-
   - email-id: k8s-infra-rbac-triageparty-cli@kubernetes.io
```
Comment (ameukam) [inline groups/groups.yaml:168]: Could be move to `groups/sig-cli` ? (the same is done for sig-release)

**Comment by ameukam** [comment]
/lgtm
/hold
If others want to comment.

File: groups/groups.yaml (hunk 1 above)
Comment (spiffxp) [reply groups/groups.yaml:123]: wasn't sure if it was simpler to leave at root or move in k8s-infra so punted for this first PR

File: groups/groups.yaml (hunk 2 above)
Comment (spiffxp) [reply groups/groups.yaml:168]: wasn't sure if this would conflict with the inflight triageparty-cli PR or not so left it for next round of PRs

**Comment by spiffxp** [comment]
I'm going to `/hold cancel` in an hour or so, none of these changes are irreversible if there are post-merge review comments.

**Pull request merged**